Editor/AGS.Types/Interfaces/IRoomController.cs

```diff
@@ -37,7 +37,8 @@ public interface IRoomController
         /// </summary>
         /// <param name="background">The background index to set.</param>
         /// <param name="bmp">The image to use for the frame.</param>
-        void SetBackground(int background, Bitmap bmp);
+        /// <param name="maskAdjustParams">The options to use for the masks when the background is changing sizes.</param>
+        void SetBackground(int background, Bitmap bmp, MaskAdjustParameters maskAdjustParams);

```

Comment:
If we ARE going to put this in `IRoomController` then it should not be a mandatory param, most people probably aren't that interested in this

```cs
public interface IRoomController
{
     void SetBackground(int background, Bitmap bmp);
     void SetBackground(int background, Bitmap bmp, MaskAdjustParameters maskAdjustParams);
}

public class RoomController : IRoomController
{
     // Map to the "real" method
     void SetBackground(int background, Bitmap bmp) => SetBackground(background, bmp, new MaskAdjustParameters { MaskOption = 0 };

     void SetBackground(int background, Bitmap bmp, MaskAdjustParameters maskAdjustParams)
     {
            // Put real implementation here
     }
}
```

Reply:
I ended up breaking out the mask resizing/resetting code from SetBackground.  So SetBackground will be unchanged and RoomSettingsEditor will call ResetMasks or ResizeMasks as needed.